OpenRA.Mods.Common/Traits/Render/WithAmmoPipsDecoration.cs

```diff
@@ -49,7 +52,9 @@ public class WithAmmoPipsDecoration : WithDecorationBase<WithAmmoPipsDecorationI
 		public WithAmmoPipsDecoration(Actor self, WithAmmoPipsDecorationInfo info)
 			: base(self, info)
 		{
-			ammo = self.TraitsImplementing<AmmoPool>().ToArray();
+			ammo = !info.AmmoPools.Any() ? self.TraitsImplementing<AmmoPool>().ToArray() :
```

Comment:
I personally would have written this as
```csharp
if (info.AmmoPools.Any())
	ammo = self.TraitsImplementing<AmmoPool>()
		.Where(ap => info.AmmoPools.Contains(ap.Info.Name))
		.ToArray();
else
	ammo = self.TraitsImplementing<AmmoPool>().ToArray();
```
which I feel is slightly more readable. Not a blocker though.